GetStream/twitter-clone
# clean up signup and login view

### What's changed

As I was navigating through the login and create account screens I noticed that the back button didn't work as I would have expected. Instead of dismissing the current popover view, it would push a new start view onto the navigation stack within the popover. I think the intention may have been for this back button to dismiss the view. Also, since the content is presented in a popover view, it would probably be better UX to provide a close or X button instead of a back button.

| Before | After |
| - | - |
| <video src='https://user-images.githubusercontent.com/1420670/231846560-873a8ba2-79bc-45c4-96ae-52ad7b5a9304.mov' width=180/> | <video src='https://user-images.githubusercontent.com/1420670/231846645-7f297ef7-b352-4083-8ff3-bf61c1581257.mov' /> |

#### Other Changes
- Renamed `Login` and `SignUp` to `LoginView` and `SignUpView` for consistency
- Moved any subviews (buttons, text fields, rows etc) within each of these main views for easier readability
```
VStack {
    titleView
    imageView
    createAccountButton
    orSeparatorView
    loginRowView
}
```
- Added a new container view `PopoverView` that reduces some of the duplication for views presented as a popover, namely wrapping the view in navigation and showing a dismiss button and login in the navigation bar.

### How to test

From the start view you can test both of the create account and login flows to see the changes as shown in the before/after videos above.
